android/app/src/main/java/org/fossasia/openevent/data/Microlocation.kt

```diff
@@ -0,0 +1,31 @@
+package org.fossasia.openevent.data
+
+import com.fasterxml.jackson.databind.PropertyNamingStrategy
+import com.fasterxml.jackson.databind.annotation.JsonNaming
+import com.github.jasminb.jsonapi.IntegerIdHandler
+import com.github.jasminb.jsonapi.annotations.Id
+import com.github.jasminb.jsonapi.annotations.Type
+
+import io.realm.RealmObject
+import io.realm.annotations.PrimaryKey
+
+@Type("microlocation")
+@JsonNaming(PropertyNamingStrategy.KebabCaseStrategy::class)
+open class Microlocation(
+        @PrimaryKey
+        @Id(IntegerIdHandler::class)
+        var id: Int = 0,
+        var name: String? = null,
+        var latitude: Float = 0.toFloat(),
+        var longitude: Float = 0.toFloat(),
+        var floor: Int = 0,
+        var room: String? = null
+
+) : RealmObject() {
+
+    constructor(id: Int, name: String) : this() {
+        this.id = id
+        this.name = name
+    }
```

Comment:
Remove extra constructor, we will used named properties in kotlin and in Java. we will manually set the id and name when creating objects